src/zarr/codecs/pipeline.py

```diff
@@ -381,6 +381,7 @@ def _merge_chunk_array(
                 out_selection: SelectorTuple,
                 chunk_spec: ArraySpec,
                 chunk_selection: SelectorTuple,
+                drop_axes: tuple[int, ...],
```

Comment:
this partially addreses my concern raised here:  https://github.com/zarr-developers/zarr-python/pull/1917/files#r1623186488